clickhouse_options.go

```diff
@@ -256,19 +263,50 @@ func (o *Options) fromDSN(in string) error {
 				}
 			}
 		case "connection_open_strategy":
-			switch params.Get(v) {
+			switch params.Get(key) {
 			case "in_order":
 				o.ConnOpenStrategy = ConnOpenInOrder
 			case "round_robin":
 				o.ConnOpenStrategy = ConnOpenRoundRobin
 			}
 		case "username":
-			o.Auth.Username = params.Get(v)
+			o.Auth.Username = params.Get(key)
 		case "password":
-			o.Auth.Password = params.Get(v)
+			o.Auth.Password = params.Get(key)
+		case "client_info_product":
+			val := params.Get(v)
+
+			if len(subKey) == 0 || len(val) == 0 {
+				continue
+			}
+
+			o.ClientInfo.Products = append(o.ClientInfo.Products, struct{ Name, Version string }{
+				subKey,
+				val,
+			})
+		case "client_info_meta":
```

Comment:
Do we need to provide `meta` and `comment` config options? The client can derive them from the env. 
What use cases do you envision?

Reply:
This will be provided in docs.

Reply:
> This will be provided in docs.

Could you elaborate, please?

Reply:
@mshustov not sure I understood your comment then. Do you mean, if it makes sense to expose an API to change meta or comment?

Reply:
>  Do you mean, if it makes sense to expose an API to change meta or comment?

I'm wondering in what cases it might be necessary for a developer to supply custom `meta` and `comment` fields. I would prefer not to introduce an API that is rarely used.

Reply:
I don't know such use cases yet. That is definitely a good point to cut it off for now until we get a clear request.